src/Preprocessing.py

```
@@ -0,0 +1,67 @@
+
+                try:
+                    img = load_and_correct_orientation(input_path)
+                    img = auto_rotate_if_landscape(img)  
+                    processed = preprocess_color_image(img)
+                    cv2.imwrite(output_path, processed)
+                    print(f"Saved: {output_path}")
+                except Exception as e:
+                    print(f"Error: {input_path} - {e}")
+
+# đường dẫn
+input_root = r"D:\Study\Semester 4\CPV\raw"
```

Flake8 (7.2.0)

[error] expected 2 blank lines after class or function definition, found 1

(E305)